libs/custom-paper-handler/src/driver/coders.ts

```diff
@@ -0,0 +1,211 @@
+import {
+  CoderType,
+  literal,
+  message,
+  uint16,
+  uint32,
+  uint8,
+} from '@votingworks/message-coder';
+import { TOKEN } from './constants';
+
+/**
+ * Response coders
+ */
+
+// Real-time exchange responses have the same fields but variable lengths of "optional" data.
+// This implementation isn't DRY but is explicit. It would be more DRY and less explicit to
+// extend message-coder to support an unbounded ArrayBuffer coder
+export const SensorStatusRealTimeExchangeResponse = message({
+  startOfPacket: literal(0x82),
+  requestId: uint8(),
+  token: literal(TOKEN),
+  returnCode: uint8(),
+  optionalDataLength: uint8(),
+  optionalByte0: uint8(),
+  optionalByte1: uint8(),
+  optionalByte2: uint8(),
+  optionalByte3: uint8(),
+});
+
+export type SensorStatusRealTimeExchangeResponse = CoderType<
+  typeof SensorStatusRealTimeExchangeResponse
+>;
+
+export const PrinterStatusRealTimeExchangeResponse = message({
+  startOfPacket: literal(0x82),
+  requestId: uint8(),
+  token: literal(TOKEN),
+  returnCode: uint8(),
+  optionalDataLength: uint8(),
+  optionalByte0: uint8(),
+  optionalByte1: uint8(),
+  optionalByte2: uint8(),
+  optionalByte3: uint8(),
+  optionalByte4: uint8(),
+  optionalByte5: uint8(),
```

Comment:
[dd0412f](https://github.com/votingworks/vxsuite/pull/3545/commits/dd0412f4f361ca5977b6e09ec3c31dfd3dac0354)

I also updated PrinterStatus to parse in the coder rather than in a separate helper according to your suggestion above but need to validate this works against the hardware